.idea/inspectionProfiles/Druid.xml

```diff
@@ -342,6 +342,9 @@
       <ignored>
         <option name="METHOD_MATCHER_CONFIG" value="java.io.PrintStream,println,java.io.PrintWriter,println,java.lang.System,identityHashCode,java.sql.PreparedStatement,set.*,java.sql.ResultSet,update.*,java.sql.SQLOutput,write.*,java.lang.Integer,compare.*,java.lang.Long,compare.*,java.lang.Short,compare,java.lang.Byte,compare,java.lang.Character,compare,java.lang.Boolean,compare,java.lang.Math,.*,java.lang.StrictMath,.*" />
       </ignored>
+      <searchConfiguration name="Prohibit Thread.getState() != TERMINATED antipattern" text="while (firehose.getDelayedCloseExecutor().getState() != Thread.State.TERMINATED) {&#10;    Thread.sleep(50);&#10;}" recursive="true" caseInsensitive="true" type="JAVA">
```

Comment:
I believe this should be inside `SSBasedInspection`. Also, it should be made more generic. This just flags one specific kind of Thread and one specific argument to Thread.sleep. I'm not sure what the correct text would be to use here, but you could try looking at the other SSBasedInspection examples for ideas.

Reply:
@gianm thanks for a quick review! made changes as per your suggestions